src/modules/iotjs_module_mqtt.c

```diff
@@ -0,0 +1,777 @@
+    case SUBSCRIBE:
+    case UNSUBSCRIBE:
+    case PINGREQ:
+      return JS_CREATE_ERROR(
+          COMMON, "MQTT: Unallowed packet has arrived to the client");
+  }
+
+  return jerry_create_undefined();
+}
+
+
+JS_FUNCTION(MqttSubscribe) {
+  DJS_CHECK_THIS();
+  DJS_CHECK_ARGS(1, object);
+
+  jerry_value_t joptions = JS_GET_ARG(0, object);
+
+  jerry_value_t jtopic =
+      iotjs_jval_get_property(joptions, IOTJS_MAGIC_STRING_TOPIC);
+  jerry_value_t jqos =
+      iotjs_jval_get_property(joptions, IOTJS_MAGIC_STRING_QOS);
+
+  uint8_t qos = 0;
+  if (!jerry_value_is_undefined(jqos) && !jerry_value_has_error_flag(jqos)) {
+    qos = (uint8_t)jerry_get_number_value(jqos);
+  }
+
+  bool dup = false;
+  bool retain = false;
+
+  uint8_t header_byte = 0;
+  header_byte |= (SUBSCRIBE << 4);
+  header_byte |= (dup << 3);
+  header_byte |= (1 << 1); // always 1
+  header_byte |= retain;
+
+  iotjs_mqtt_payload_t topic_payload =
+      iotjs_create_payload(iotjs_bufferwrap_from_jbuffer(jtopic));
+
+  // IF NOT QOS
+  uint8_t packet_identifier_lsb = 0; // TODO: This must be a changing number.
+  uint8_t packet_identifier_msb = 0;
+
+  size_t payload_len = sizeof(qos) + topic_payload.size;
+  size_t variable_header_len =
+      sizeof(packet_identifier_msb) + sizeof(packet_identifier_lsb);
+  uint32_t remaining_length = payload_len + variable_header_len;
+  size_t full_len = sizeof(header_byte) + payload_len + variable_header_len +
+                    get_remaining_length_size(remaining_length);
+
+  jerry_value_t jbuff = iotjs_bufferwrap_create_buffer(full_len);
+  iotjs_bufferwrap_t *buffer_wrap = iotjs_bufferwrap_from_jbuffer(jbuff);
+
+  uint8_t *buff_ptr = (uint8_t *)buffer_wrap->buffer;
+
+  *buff_ptr++ = header_byte;
+
+  buff_ptr += iotjs_encode_remaining_length(remaining_length, buff_ptr);
+
+  *buff_ptr++ = packet_identifier_msb;
+  *buff_ptr++ = packet_identifier_lsb;
+
+  *buff_ptr++ = topic_payload.msb;
+  *buff_ptr++ = topic_payload.lsb;
+  memcpy(buff_ptr, topic_payload.buffer, topic_payload.buffer_size);
+  buff_ptr += topic_payload.buffer_size;
+
+  *buff_ptr++ = qos;
+
+  jerry_release_value(jtopic);
+  jerry_release_value(jqos);
+
+  return jbuff;
+}
+
+
+JS_FUNCTION(MqttPing) {
+  DJS_CHECK_THIS();
+
+  uint8_t header_byte = 0;
+  header_byte |= (PINGREQ << 4);
+
+  uint8_t remaining_length = 0;
+  size_t full_len = sizeof(header_byte) + sizeof(remaining_length);
+
+  jerry_value_t jbuff = iotjs_bufferwrap_create_buffer(full_len);
+  iotjs_bufferwrap_t *buffer_wrap = iotjs_bufferwrap_from_jbuffer(jbuff);
+
+  uint8_t *buff_ptr = (uint8_t *)buffer_wrap->buffer;
+
+  *buff_ptr++ = header_byte;
+  *buff_ptr++ = remaining_length;
+
+  return jbuff;
+}
+
+
+JS_FUNCTION(MqttUnsubscribe) {
+  DJS_CHECK_THIS();
+  DJS_CHECK_ARGS(1, object);
+
+  jerry_value_t jtopic = JS_GET_ARG(0, object);
+
+  iotjs_mqtt_payload_t topic_payload =
+      iotjs_create_payload(iotjs_bufferwrap_from_jbuffer(jtopic));
+
+  uint8_t header_byte = 0;
```

Comment:
`uint8_t header_byte = (UNSUBSCRIBE << 4);`
No need every step in separeted lines.